File: LO52_F1_Levier/app/src/main/java/com/mguichar/lo52_f1/EquipeDao.java
package com.mguichar.lo52_f1;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Transaction;

import java.util.List;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface EquipeDao {

    @Insert(onConflict = REPLACE)
    void insert(Equipe equipe);

    @Delete
    void reset(List<Equipe> equipes);

    @Query("SELECT * FROM equipes")
    List<Equipe> getAll();

    @Query("SELECT * FROM equipes WHERE nom = :nom")
    List<Equipe> getEquipeByName(String nom);

}

File: CourseCODEP25/app/src/main/java/com/utbm/nouassi/manou/coursecodep25/database/dao/EquipeDao.kt
package com.utbm.nouassi.manou.coursecodep25.database.dao

import androidx.room.*
import com.utbm.nouassi.manou.coursecodep25.database.entity.Equipe
import com.utbm.nouassi.manou.coursecodep25.database.entity.EquipeAndCoureur

@Dao
interface EquipeDao {
    @Insert
    fun insertAll(vararg equipes: Equipe)

    @Insert
    fun insert(equipe: Equipe): Long

    @Delete
    fun delete(equipe: Equipe)

    @Update
    fun updateEquipes(vararg equipes: Equipe)

    @Transaction
    @Query("SELECT * FROM equipe")
    fun getAll(): List<EquipeAndCoureur>

    @Transaction
    @Query("SELECT * FROM equipe where courseId = :id")
    fun getByCourseId(id: Int): List<EquipeAndCoureur>

}